### ISSUE TITLE
Add splitlines?

### ISSUE THREAD

**Issue by mpastell** [JuliaLang/julia#19759]
I think it would be useful to have a `splitlines` method similar to Python

> Return a list of the lines in the string, breaking at line boundaries. Line breaks are not included in the resulting list unless keepends is given and true.

https://docs.python.org/3/library/stdtypes.html?highlight=splitlines .

I think it would be convenient, at least for me I very often process strings line by line. And it would be useful in order to avoid bugs related to `\r`on Windows.

**Comment by oscardssmith** [comment]
Possibly more useful would be a way to iterate by line, as split lines only works well for small files. (Both would be cool too)

**Comment by mpastell** [comment]
There is `eachline`, but it keeps the line endings and doesn't currently work for in memory strings. I think adding `splitlines` in addition to that would be useful. And give `eachline` the option not to include line breaks in the output.

**Comment by giordano** [comment]
What is the difference with [`split`](http://docs.julialang.org/en/latest/stdlib/strings.html#Base.split)?

**Comment by mpastell** [comment]
The difference to to split is automatically handling `\r\n` and `\n` as linefeed. The first implementation that comes to mind would be:

```julia
split(string, r"\n|\r\n")
```
The corresponding function in Python also handles several other linefeed formats and it could be wise to do the same. I recently had few bugs in Weave.jl due to not handling `\r`(I just did `split(s, "\n")` ) and I suspect that others not using Julia on Windows regularly might do the same.

Using `rstrip` when parsing formats with significant whitespace is not a good idea. There is also `chomp`, but it doesn't remove trailing `\r`.

**Comment by giordano** [comment]
I'd say that there is no need for a new function (`split` is just fine), but to improve handling of newlines (and this is the topic of https://github.com/JuliaLang/julia/issues/19785), right?  If so, this ticket can be closed.

**Comment by mpastell** [comment]
I think `splitlines` contributes to improved handling of newlines and it would be nice to have it similarly as you have both `print` and `println`, `readlines` and `eachline` etc. Many languages have both `split` and `splitlines` e.g. Python, Ruby and Clojure.

**Comment by giordano** [comment]
I see your point now, thanks for clarifying it.

**Comment by mpastell** [comment]
After #19944 gets merged this could simply be defined as 

```julia
splitlines(str::String, chomp::Bool=true) = readlines(IOBuffer(str), chomp))
```

**Comment by laborg** [comment]
This issue can be closed: `eachline` and `readline` have a `keep` keyword since https://github.com/JuliaLang/julia/pull/20203 was merged.
